src/components/ChainLogo.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import styled from 'styled-components';
+import ArbitrumBadge from '../assets/badges/arbitrumBadge.png';
+import ArbitrumBadgeDark from '../assets/badges/arbitrumBadgeDark.png';
+import ArbitrumBadgeNoShadow from '../assets/badges/arbitrumBadgeNoShadow.png';
+import OptimismBadge from '../assets/badges/optimismBadge.png';
+import OptimismBadgeDark from '../assets/badges/optimismBadgeDark.png';
+import OptimismBadgeNoShadow from '../assets/badges/optimismBadgeNoShadow.png';
+import PolygonBadge from '../assets/badges/polygonBadge.png';
+import PolygonBadgeDark from '../assets/badges/polygonBadgeDark.png';
+import PolygonBadgeNoShadow from '../assets/badges/polygonBadgeNoShadow.png';
+import { ImgixImage } from './images';
+import { Centered } from './layout';
+import networkTypes from '@rainbow-me/helpers/networkTypes';
+import { position } from '@rainbow-me/styles';
+
+const ChainIcon = styled(ImgixImage)`
+  height: ${({ size }) => size};
+  margin-top: 1;
+  width: ${({ size }) => size};
+`;
+
+const Content = styled(Centered)`
+  ${({ size }) => position.size(size / 2)};
+  overflow: visible;
+`;
+
+export default function ChainLogo({
+  network,
+  size = 40,
+  withShadows = true,
+  ...props
+}) {
+  const { isDarkMode } = useTheme();
+  const source = useMemo(() => {
+    let val = null;
+    if (network === networkTypes.arbitrum) {
+      val = withShadows
+        ? isDarkMode
+          ? ArbitrumBadgeDark
+          : ArbitrumBadge
+        : ArbitrumBadgeNoShadow;
+    } else if (network === networkTypes.optimism) {
+      val = val = withShadows
```

Comment:
what's this with `val = val = withShadows` ?

Reply:
🙈